src/components/elements/table/RestockTableMain.tsx

```diff
@@ -0,0 +1,214 @@
+              direction={orderBy === headCell.id ? order : 'asc'}
+              onClick={createSortHandler(headCell.id)}
+            >
+              {headCell.label}
+              {orderBy === headCell.id ? (
+                <span className={classes.visuallyHidden}>
+                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                </span>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>,
+
+        )}
+        <TableCell>{'Purchase Info'}</TableCell>
+      </TableRow>
+    </TableHead>
+  );
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      width: '100%',
+    },
+    paper: {
+      width: '100%',
+      marginBottom: theme.spacing(2),
+    },
+    table: {
+      minWidth: 750,
+    },
+    visuallyHidden: {
+      border: 0,
+      clip: 'rect(0 0 0 0)',
+      height: 1,
+      margin: -1,
+      overflow: 'hidden',
+      padding: 0,
+      position: 'absolute',
+      top: 20,
+      width: 1,
+    },
+  }),
+);
+
+export const RestockPurchasesTable = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const {restockData} = useRestockSelector();
+
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderBy, setOrderBy] = React.useState<keyof RestockDataInfo>('id');
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(10);
+
+  const getData = () => {
+    dispatch(restockDispatchers.fetchRestockData());
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+
+  // TODO: FIGURE OUT IF THIS SHOULD BE A REDUX ACTION OR NOT
+  const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof RestockDataInfo) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  let emptyRows = 0;
+  if (restockData != null) {
+    emptyRows = rowsPerPage - Math.min(rowsPerPage, restockData.length - page * rowsPerPage);
+  }
+
+  return (
+    <div className={classes.root}>
+      {
+        restockData != null ?
+          <Paper className={classes.paper}>
+            <TableContainer>
+              <Table
+                className={classes.table}
+                aria-labelledby="tableTitle"
+                aria-label="enhanced table"
+              >
+                <EnhancedTableHead
+                  classes={classes}
+                  order={order}
+                  orderBy={orderBy}
+                  onRequestSort={handleRequestSort}
+                />
+                <TableBody>
+                  {stableSort(restockData, getComparator(order, orderBy))
```

Comment:
Getting rid of comparator and have something like `sort(restockData, order, (item) => item[orderBy])` equivalent to this call will looks better to me (and also getting rid of the complicating comparator code in sorting utils)

Reply:
I don't feel comfortable changing this. This was the built-in sorting that Material-UI had. You can change it if you feel it's necessary.